airflow/providers/databricks/sensors/databricks_partition.py

```diff
@@ -0,0 +1,157 @@
+        as HTTP headers on every request. (templated)
+    :param catalog: An optional initial catalog to use. Requires DBR version 9.0+ (templated)
+    :param schema: An optional initial schema to use. Requires DBR version 9.0+ (templated)
+    :param table_name: Table name to generate the SQL query.
+    :param partition_name: Partition to check.
+    :param handler: Handler for DbApiHook.run() to return results, defaults to fetch_one_handler
+    :param caller: String passed to name a hook to Databricks, defaults to "DatabricksPartitionSensor"
+    :param client_parameters: Additional parameters internal to Databricks SQL Connector parameters.
+    :param partition_operator: Comparison operator for partitions.
+    """
+
+    template_fields: Sequence[str] = (
+        "databricks_conn_id",
+        "schema",
+        "http_headers",
+        "catalog",
+        "table_name",
+        "partition_name",
+    )
+
+    def __init__(
+        self,
+        *,
+        databricks_conn_id: str = DatabricksSqlHook.default_conn_name,
+        http_path: str | None = None,
+        sql_endpoint_name: str | None = None,
+        session_configuration=None,
+        http_headers: list[tuple[str, str]] | None = None,
+        catalog: str = "",
+        schema: str = "default",
+        table_name: str = "",
+        partition_name: dict,
+        handler: Callable[[Any], Any] = fetch_one_handler,
+        caller: str = "DatabricksPartitionSensor",
+        client_parameters: dict[str, Any] | None = None,
+        partition_operator: str = "=",
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.databricks_conn_id = databricks_conn_id
+        self._http_path = http_path
+        self._sql_endpoint_name = sql_endpoint_name
+        self.session_config = session_configuration
+        self.http_headers = http_headers
+        self.catalog = catalog
+        self.schema = schema
+        self.table_name = table_name
+        self.partition_name = partition_name
+        self.caller = caller
+        self.client_parameters = client_parameters or {}
+        self.hook_params = kwargs.pop("hook_params", {})
+        self.handler = handler
+        self.partition_operator = partition_operator
+
+    def _get_hook(self) -> DatabricksSqlHook:
+        return DatabricksSqlHook(
+            self.databricks_conn_id,
+            self._http_path,
+            self._sql_endpoint_name,
+            self.session_config,
+            self.http_headers,
+            self.catalog,
+            self.schema,
+            self.caller,
+            **self.client_parameters,
+            **self.hook_params,
+        )
+
+    def _sql_sensor(self, sql):
+        hook = self._get_hook()
+        sql_result = hook.run(
+            sql,
+            handler=self.handler if self.do_xcom_push else None,
+        )
+        return sql_result
+
+    def _check_table_partitions(self) -> list:
+        if self.catalog is not None:
```

Comment:
it will be always executed this way because we're passing empty string as default. Also, we can continue to rely on two-level and even one level naming, relying on default catalog & default schema...

Reply:
Changed it.